packages/@lwc/synthetic-shadow/src/faux-shadow/traverse.ts

```diff
@@ -134,24 +134,33 @@ export function isSlotElement(node: Node): node is HTMLSlotElement {
     return node instanceof HTMLSlotElement;
 }
 
-export function isNodeOwnedBy(owner: Element, node: Node): boolean {
+export function isNodeOwnedBy(owner: Element | null, node: Node): boolean {
     if (process.env.NODE_ENV !== 'production') {
-        if (!(owner instanceof HTMLElement)) {
-            // eslint-disable-next-line no-console
-            console.error(`isNodeOwnedBy() should be called with an element as the first argument`);
-        }
         if (!(node instanceof Node)) {
             // eslint-disable-next-line no-console
             console.error(`isNodeOwnedBy() should be called with a node as the second argument`);
         }
+    }
+
+    // owner can be null if all the parents are in the light DOM
+    // ownerKey is undefined for such nodes
+    const ownerKey = getNodeNearestOwnerKey(node);
+    if (isNull(owner)) {
+        return isUndefined(ownerKey);
```

Comment:
Not so sure about this logic. If the owner is `null` but `ownerKey` is defined, then `isNodeOwnedBy()` returns `true`. By relaxing the input constraints, it seems that we're muddying the job of this function. Maybe this is a sign that we should be handling the light DOM case elsewhere?

Reply:
I think of this as node being owned by a shadowRoot or document (in which case it's null). It's owned by document, in case of light DOM parents.

Reply:
Would having more tests for childNodes/children help build confidence? I suspect we don’t have a lot.

Reply:
For both `childrenGetterPatched` and `childNodesGetterPatched`, if there is no owner, then it means that the element or node is not shadowed, correct? If so, then it seems like we should immediately return `getFilteredChildNodes(this)` instead of drilling down into `getAllMatches()` and `isNodeOwnedBy()`.

Reply:
Makes sense! Done.